java/arcs/core/host/ArcState.kt

```diff
@@ -134,7 +134,7 @@ data class ParticleState private constructor(val state: State) {
 
     override fun toString(): String {
         return if (_cause != null) {
-            "${state.name}|$_cause"
+            "${state.name}|${cause!!.message}"
```

Comment:
Can this be structured in a way that the non-null assertion isn't needed?

Also curious if we need `_cause` vs `cause`. Isn't it possible to do something like
```
var cause: Exception?
  private set
```

Reply:
Probably, the type checker won't let the null assertion be removed, it claims the if(_cause != null) can't guard the second reference (and changing to use _cause.message doesn't help)